src/router_node.c

```diff
@@ -329,12 +330,32 @@ static bool AMQP_rx_handler(void* context, qd_link_t *link)
     qdr_delivery_t *delivery = qdr_node_delivery_qdr_from_pn(pnd);
     bool       next_delivery = false;
 
-    //
-    // Receive the message into a local representation.
-    //
     qd_message_t   *msg   = qd_message_receive(pnd);
+    
+    bool is_oversize      = qd_message_exceeds_max_message_size(msg);
+    if (is_oversize) {
```

Comment:
Is there a case when is_oversize is true but qd_message_oversize_detected(msg) is false ?